thunderfish/sortfishtype.py

```diff
@@ -0,0 +1,189 @@
+import numpy as np
+import peakdetection as pkd
+from IPython import embed
+
+
+def type_detector(data, samplerate, thres=0.1, plot_data_func=None):
+
+    """ Detects if fish is pulse or wave by calculating the proportion of the time distance between a peak and its
+     following trough, relative to the time between consecutive peaks.
+
+    :param data: (1-D array) The data to be analyzed (Usually the best window already)
+    :param samplerate: (float). Sampling rate of the data in Hz
+    :param thres: (float or array): a positive number setting the minimum distance between peaks and troughs
+    :param plot_data_func: Function for plotting the raw data, detected peaks and troughs, the criteria,
+    the cost function and the selected best window.
+        plot_data_func(data, rate, peak_idx, trough_idx, idx0, idx1,
+                       win_start_times, cv_interv, mean_ampl, cv_ampl, clipped_frac, cost,
+                       thresh, valid_wins, **kwargs)
+        :param data (array): the raw data.
+        :param rate (float): the sampling rate of the data.
+        :param peak_idx (array): indices into raw data indicating detected peaks.
+        :param trough_idx (array): indices into raw data indicating detected troughs.
+    :return: suggestion: (str). Returns the suggested fish-type ("pulse"/"wave").
+    :return: r_value: (float). Returns a float between 0. and 1. which gives the proportion of peak-2-trough,
+                            from peak-2-peak time distance. (Wave-fishes should have larger values than pulse-fishes)
+    """
+
+    print('\nAnalyzing Fish-Type...')
+
+    # define args for peak-trough detection
+    win_shift = 0.1
+
+    # threshold for peak detection:
+    threshold = np.zeros(len(data))
+    win_shift_indices = int(win_shift * samplerate)
+
```

Comment:
We could discuss whether this function expects a stable piece of data. Then it would be sufficient to compute a single threshold for the whole data vector. This would also spare the win_shift parameter.